src/LanguageServer/Protocol/Features/Diagnostics/EngineV2/DiagnosticIncrementalAnalyzer.Executor.cs

```diff
@@ -23,329 +23,218 @@ private partial class DiagnosticIncrementalAnalyzer
         /// <summary>
         /// Return all diagnostics that belong to given project for the given StateSets (analyzers) either from cache or by calculating them
         /// </summary>
-        private async Task<ProjectAnalysisData> GetProjectAnalysisDataAsync(
+        private async Task<ImmutableDictionary<DiagnosticAnalyzer, DiagnosticAnalysisResult>> ComputeDiagnosticAnalysisResultsAsync(
             CompilationWithAnalyzersPair? compilationWithAnalyzers, Project project, ImmutableArray<StateSet> stateSets, CancellationToken cancellationToken)
         {
             using (Logger.LogBlock(FunctionId.Diagnostics_ProjectDiagnostic, GetProjectLogMessage, project, stateSets, cancellationToken))
             {
                 try
                 {
-                    var version = await GetDiagnosticVersionAsync(project, cancellationToken).ConfigureAwait(false);
```

Comment:
'versions' ended up being virtually eliminated.  GetDiagnosticVersionAsync returns project.GetDependentVersionAsync which is:

> The most recent version of the project, its documents and all dependent projects and documents.

So any change to eh project at all (options, info, docs, dependencies) would rev this.  no need for this at all now, since we can just use the Project instnace itself in a CWT.